### PR TITLE
global: --with-universal-ctags

### PR DESCRIPTION
**Pull request by ivanbrennan**
GNU Global supports using Universal Ctags and Exuberant Ctags as plug-in parsers, but the current formula only exposes the Exuberant Ctags option.

This is problematic, as Universal Ctags is the actively maintained/developed implementation. The Global documentation describes how to configure Universal Ctags and makes no mention of Exuberant Ctags.

Expose `--with-universal-ctags` option to remedy this.

To install:

    $ brew install global --with-universal-ctags

To use:

    $ export GTAGSCONF=/usr/local/share/gtags/gtags.conf
    $ export GTAGSLABEL=new-ctags
    $ gtags                         # gtags invokes Universal Ctags internally

References:
  https://www.gnu.org/software/global/manual/global.html#Plug_002din
  https://github.com/universal-ctags/homebrew-universal-ctags

### CODE REVIEW

**Comment by ivanbrennan** [comment]
Tests are currently failing because the formula can't find the `universal-ctags` formula (I ran tests locally, but only after running `brew tap universal-ctags/universal-ctags`.

I'm looking into either auto-tapping or providing an informative error in such cases -- not sure which is the preferred behavior.

**Comment by ivanbrennan** [comment]
I updated the `depends_on` call to correctly reference the tapped `universal-ctags` formula:
```diff
 -  depends_on "universal-ctags" => :optional
 +  depends_on "universal-ctags/universal-ctags/universal-ctags" => :optional
```

File: Formula/global.rb
Comment (MikeMcQuaid) [inline Formula/global.rb]: We don't want homebrew/core formulae depending on non-Homebrew organisation taps as options. Could this be submitted to homebrew/core?

File: Formula/global.rb
Comment (MikeMcQuaid) [inline Formula/global.rb]: Don't need this unless you modify the non-option installation.

File: Formula/global.rb
Comment (MikeMcQuaid) [inline Formula/global.rb]: Probably not worth adding this; these tests are only for CI and thus will never be run with options so this code will never be run.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/global.rb
Comment (ivanbrennan) [inline Formula/global.rb]: I wasn't sure whether exposing a new option warranted/required adding a `revision`. I'm starting to suspect not, but I'll leave this for now to get some outside feedback.

Comment (MikeMcQuaid) [inline Formula/global.rb]: It definitely doesn't so you can remove it, thanks. A `revision` will force a reinstall and, in some cases, rebuild for everyone using this.

File: Formula/global.rb
Comment (ivanbrennan) [inline Formula/global.rb]: @MikeMcQuaid The [universal-ctags tap](https://github.com/universal-ctags/homebrew-universal-ctags) mentions,

> When Universal Ctags has a tagged release, this formula should be moved to the official Homebrew repository. This repo should then be deleted.

Indeed, [universal ctags](https://github.com/universal-ctags/ctags) still has no tagged release. I'll check with the maintainers on the status of that. Is a tagged release a prerequisite for submitting to homebrew/core?

Comment (MikeMcQuaid) [inline Formula/global.rb]: It is, yep.

**Comment by dunn** [comment]
Do you have any idea when a stable release is expected?

**Comment by ivanbrennan** [comment]
@dunn I found multiple universal ctags issues discussing their plans for a release, but it all seems up in the air at the moment. For reference:
https://github.com/universal-ctags/ctags/issues/1038
https://github.com/universal-ctags/ctags/issues/664
https://github.com/universal-ctags/ctags/issues/658
https://github.com/universal-ctags/ctags/issues/909
https://github.com/universal-ctags/ctags/issues/662
https://github.com/universal-ctags/ctags/issues/652

Apologies for the barrage of links, maybe too much information, but that's what I found.

**Comment by dunn** [comment]
Thanks!  If there's no definite ETA, you can still leave this PR open, but we're gonna want that stable release before merging.

**Comment by ivanbrennan** [comment]
Sounds good. Thanks @dunn

**Comment by MikeMcQuaid** [comment]
Closing until we've got a stable release.

**Pull request closed without merging**
